src/map/battlefield_handler.cpp

```diff
@@ -602,104 +405,43 @@ uint8 CBattlefieldHandler::findBattlefieldIDFor(CCharEntity* PChar) {
     return 255;
 }
 
-CBattlefield* CBattlefieldHandler::getBattlefield(CCharEntity* PChar)
+CBattlefield* CBattlefieldHandler::getBattlefield(uint16 bnum)
 {
-    for (int i = 0; i < m_MaxBattlefields; i++)
-        if (m_Battlefields[i] != nullptr)
-            if (m_Battlefields[i]->isValidPlayerForBcnm(PChar))
-                return m_Battlefields[i];
-    return nullptr;
+    return m_Battlefields[bnum-1];
 }
 
-uint32 CBattlefieldHandler::pollTimeLeft(uint16 id) {
-    return 0;
-}
+int CBattlefieldHandler::addPlayerToBattlefield(uint16 bnum, CCharEntity* PChar) {
 
-void CBattlefieldHandler::openTreasureChest(CCharEntity* PChar) {
-    for (int i = 0; i < m_MaxBattlefields; i++) {
-        if (m_Battlefields[i] != nullptr) {
-            if (m_Battlefields[i]->isValidPlayerForBcnm(PChar)) {
-                CBattlefield* PBattlefield = m_Battlefields[i];
-                PBattlefield->OpenChestinBcnm();
-            }
-        }
+    if (m_Battlefields[bnum-1]->addPlayer(PChar)) {
+        ShowDebug("Zone %i battlefield %i with bfid %i: Added %s \n", m_Battlefields[bnum-1]->getZoneId(), m_Battlefields[bnum-1]->getBattlefieldNumber(), m_Battlefields[bnum-1]->getID(), PChar->GetName());
     }
-}
 
-//========================DYNAMIS FUNCTIONS=============================================//
+    return 1;
+}
 
-int CBattlefieldHandler::getUniqueDynaID(uint16 id) {
 
-    CBattlefield* PBattlefield = m_Battlefields[0];
-    return PBattlefield->getDynaUniqueID();
-}
+//========================DYNAMIS FUNCTIONS=============================================//
 
-int CBattlefieldHandler::registerDynamis(uint16 id, CCharEntity* PChar) {
-    if (!hasFreeBattlefield()) {
+int CBattlefieldHandler::registerDynamis(uint16 bnum, uint16 id, CCharEntity* PChar) {
```

Comment:
ideally dynamis functions would be removed entirely, since there's no need for them besides spawning next wave of mobs (so probably wanna handle tick in lua for special snowflakes like dyna and temenos)

Reply:
removed registerDynamis in further commit